### PR TITLE
Remove the global partition code from hypre

### PR DESCRIPTION
**Pull request by rfalgout**
This PR removes the global partition code from hypre.  There is little reason to keep it anymore.

### CODE REVIEW

File: src/parcsr_ls/par_mgr_setup.c [hunk 1]
```diff
@@ -1406,12 +1406,8 @@ hypre_MGRSetupFrelaxVcycleData( void *mgr_vdata,
                         num_functions, dof_func_array[lev_local], CF_marker_local,
                         &coarse_dof_func_lvl, &coarse_pnts_global_lvl);
 
-#ifdef HYPRE_NO_GLOBAL_PARTITION
     if (my_id == (num_procs -1)) coarse_size = coarse_pnts_global_lvl[1];
       hypre_MPI_Bcast(&coarse_size, 1, HYPRE_MPI_BIG_INT, num_procs-1, comm);
```
Comment (liruipeng) [inline src/parcsr_ls/par_mgr_setup.c:1410]: @rfalgout @oseikuffuor1 This is not the focus of this PR. but the indentation of this line (`hypre_MPI_Bcast`) is confusing.

**Review by liruipeng — APPROVED** [review]
Looks good to me. Thank you, Rob!

File: src/parcsr_ls/par_mgr_setup.c (hunk 1 above)
Comment (oseikuffuor1) [reply src/parcsr_ls/par_mgr_setup.c:1410]: @liruipeng I can fix this in PR #260, thanks.

**Review by ulrikeyang — APPROVED** [review]
Looks good.

**Review by oseikuffuor1 — APPROVED** [review]
Looks good to me too. Thanks

**Pull request merged**